Nanoleaf-wpf/Views/App.xaml.cs

```diff
@@ -141,5 +144,28 @@ public static void ResetAllSettings(MainWindow mainWindow)
 
             mainWindow.Close();
         }
+
+        private static void CheckForUpdate()
+        {
+            // check if the user has done the setup yet, if not we assume he has the latest version for now.
```

Comment:
If we decouple as said in my previous comment, then this check is not needed anymore